src/flamenco/snapshot/fd_snapshot_create.c

```diff
@@ -0,0 +1,1224 @@
+#include "fd_snapshot_create.h"
+#include "../runtime/sysvar/fd_sysvar_epoch_schedule.h"
+#include "../../ballet/zstd/fd_zstd.h"
+#include "../runtime/fd_hashes.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <zstd.h>
+
+static uchar padding [ FD_SNAPSHOT_ACC_ALIGN ] = {0};
+
+static inline int
+fd_snapshot_create_populate_acc_vecs( fd_snapshot_ctx_t                 * snapshot_ctx,
+                                      fd_solana_manifest_serializable_t * manifest,
+                                      fd_tar_writer_t                   * writer,
+                                      ulong                             * out_cap ) {
+
+  /* The append vecs need to be described in an index in the manifest so a
+     reader knows what account files to look for. These files are technically
+     slot indexed, but the Firedancer implementation of the Solana snapshot
+     produces far fewer indices. These storages are for the accounts 
+     that were modified and deleted in the most recent slot because that 
+     information is used by the Agave client to calculate and verify the 
+     bank hash for the given slot. This is done as an optimization to avoid
+     having to slot index the Firedancer accounts db which would incur a large
+     performance hit.
+     
+     To avoid iterating through the root twice to determine what accounts were
+     touched in the snapshot slot and what accounts were touched in the
+     other slots, we will create an array of pubkey pointers for all accounts
+     that were touched in the pubkey slot. This buffer can be safely sized to 
+     the maximum amount of writable accounts that are possible in a non-epoch
+     boundary slot. The rationale for this bound is explained in
```

Comment:
Let's not make this happen. I'll add casing around this. It could potentially mess up some invariants around sizes of append vecs on Agave's side. I'll update s.t. we don't try to create on an epoch boundary.